### PR TITLE
Add the set_interface_mtu compliance test

### PR DESCRIPTION
**Pull request by lindycoder**
Along with changes to the api, remote and cached layers
Get interface for juniper was also modified to return the
value

### CODE REVIEW

File: netman/adapters/switches/cached.py [hunk 1]
```diff
@@ -380,4 +380,20 @@ def set_vlan_icmp_redirects_state(self, vlan_number, state):
     def get_versions(self):
         if self.versions_cache.refresh_items:
             self.versions_cache = Cache([(0, self.real_switch.get_versions())])
-        return copy.deepcopy(self.versions_cache[0])        
+        return copy.deepcopy(self.versions_cache[0])
+
+    def set_interface_mtu(self, interface_id, vlan):
```
Comment (idjaw) [inline netman/adapters/switches/cached.py]: s/vlan/size/ no?

File: netman/adapters/switches/cached.py [hunk 2]
```diff
@@ -380,4 +380,20 @@ def set_vlan_icmp_redirects_state(self, vlan_number, state):
...
+        self.real_switch.set_interface_mtu(interface_id, vlan)
+        self.interfaces_cache[interface_id].mtu = vlan
+
+    def unset_interface_mtu(self, interface_id):
+        self.real_switch.unset_interface_mtu(interface_id)
+        self.interfaces_cache[interface_id].mtu = None
+
+    def set_bond_mtu(self, bond_number, vlan):
```
Comment (idjaw) [inline netman/adapters/switches/cached.py]: s/vlan/size/

File: netman/adapters/switches/cached.py (hunk 1 above)
Comment (lindycoder) [inline netman/adapters/switches/cached.py]: yes thanks

File: netman/adapters/switches/remote.py [hunk 3]
```diff
@@ -226,6 +226,18 @@ def set_bond_description(self, bond_number, description):
     def unset_bond_description(self, bond_number):
         self.delete("/bonds/" + str(bond_number) + '/description')
 
+    def set_interface_mtu(self, interface_id, mtu):
```
Comment (idjaw) [inline netman/adapters/switches/remote.py]: The switch_base uses size instead of mtu. Should we stay consistent and keep it the same var name throughout?

Comment (lindycoder) [inline netman/adapters/switches/remote.py]: yes

**Comment by lindycoder** [comment]
Comments have been addressed

File: netman/adapters/switches/remote.py [hunk 4]
```diff
@@ -226,6 +226,18 @@ def set_bond_description(self, bond_number, description):
     def unset_bond_description(self, bond_number):
         self.delete("/bonds/" + str(bond_number) + '/description')
 
+    def set_interface_mtu(self, interface_id, size):
+        self.put("/interfaces/" + interface_id + '/mtu', raw_data=str(size))
+
+    def unset_interface_mtu(self, interface_id):
+        self.delete("/interfaces/" + interface_id + '/mtu')
+
+    def set_bond_mtu(self, bond_number, mtu):
```
Comment (idjaw) [inline netman/adapters/switches/remote.py]: consistency. mtu should be size.

Comment (lindycoder) [inline netman/adapters/switches/remote.py]: done

**Comment by idjaw** [comment]
LGTM 👍

**Comment by godp1301** [comment]
👍

**Pull request merged**